src/test/java/cz/fi/muni/pa165/esports/dao/PlayerDaoTest.java

```diff
@@ -0,0 +1,199 @@
+
+    @Test
+    public void testDelete() {
+        // using DAO as find and delete have to be in the same entity manager
+        Player foundPlayer = playerDao.findById(mrBlonde.getPlayer_id());
+        playerDao.delete(foundPlayer);
+
+        foundPlayer = playerDao.findById(mrBlonde.getPlayer_id());
+        Assert.assertNull(foundPlayer);
+    }
+
+    @Test
+    public void testFindByIdNotFound() {
+        Player foundPlayer = playerDao.findById(mrBlonde.getPlayer_id());
+        Assert.assertNull(foundPlayer);
+    }
+
+    @Test
+    public void testFindAll() {
+        List<Player> foundPlayers = playerDao.findAll();
+        Assert.assertNotNull(foundPlayers);
+        Assert.assertEquals(foundPlayers.size(), 3);
+        
+        assertContainsPlayerWithName(foundPlayers, mrWhite.getName());
+        assertContainsPlayerWithName(foundPlayers, mrOrange.getName());
+        assertContainsPlayerWithName(foundPlayers, mrsRed.getName());
+    }
+
+    @Test
+    public void testFindByNameFoundSingle() {
+        List<Player> foundPlayers = playerDao.findByName(mrOrange.getName());
+        Assert.assertNotNull(foundPlayers);
+        Assert.assertEquals(foundPlayers.size(), 1);
+        assertContainsPlayerWithName(foundPlayers, mrOrange.getName());
+    }
+
+    @Test
+    public void testFindByNameFoundMultiple() {
+        List<Player> foundPlayers = playerDao.findByName(mrWhite.getName());
+        Assert.assertNotNull(foundPlayers);
+        Assert.assertEquals(foundPlayers.size(), 2);
+        assertContainsPlayerWithName(foundPlayers, mrWhite.getName());
+    }
+    
+    @Test
+    public void testFindByNameNotFound() {
+        List<Player> foundPlayers = playerDao.findByName(mrBrown.getName());
+        Assert.assertNotNull(foundPlayers);
+        Assert.assertEquals(foundPlayers.size(), 0);
+    }
+
+    @Test
+    public void testFindByGenderFoundSingle() {
+        List<Player> foundPlayers = playerDao.findByGender(mrsRed.getGender());
+        Assert.assertNotNull(foundPlayers);
+        Assert.assertEquals(foundPlayers.size(), 1);
+    }
+
+    @Test
+    public void testFindByGenderFoundMultiple() {
+        List<Player> foundPlayers = playerDao.findByGender(Gender.MALE);
+        Assert.assertNotNull(foundPlayers);
+        Assert.assertEquals(foundPlayers.size(), 2);
+    }
+    
+    @Test
+    public void testFindByGenderNotFound() {
+        List<Player> foundPlayers = playerDao.findByGender(Gender.OTHER);
+        Assert.assertNotNull(foundPlayers);
+        Assert.assertEquals(foundPlayers.size(), 0);
+    }
+
+    private void assertContainsPlayerWithName(List<Player> players, String expectedPlayerName) {
+        for (Player player: players) {
+            if (player.getName().equals(expectedPlayerName)) {
+                return;
+            }
+        }
+        Assert.fail("Couldn't find player " + expectedPlayerName + " in collection " + players);
+    }
+
+    @AfterClass
+    private void cleanUp() {
```

Comment:
is this needed?

Reply:
the `cleanUp()` method? I create an entity manager manually in my tests, so I wanted to make sure and close it manually as well, but I'm not sure to be honest.